source/base/function_parser.cc

```diff
@@ -330,48 +207,12 @@ FunctionParser<dim>::init_muparser() const
           // we may find after function names
           std::string transformed_expression = expressions[component];
 
-          const char *function_names[] = {// functions predefined by muparser
-                                          "sin",
-                                          "cos",
-                                          "tan",
-                                          "asin",
-                                          "acos",
-                                          "atan",
-                                          "sinh",
-                                          "cosh",
-                                          "tanh",
-                                          "asinh",
-                                          "acosh",
-                                          "atanh",
-                                          "atan2",
-                                          "log2",
-                                          "log10",
-                                          "log",
-                                          "ln",
-                                          "exp",
-                                          "sqrt",
-                                          "sign",
-                                          "rint",
-                                          "abs",
-                                          "min",
-                                          "max",
-                                          "sum",
-                                          "avg",
-                                          // functions we define ourselves above
-                                          "if",
-                                          "int",
-                                          "ceil",
-                                          "cot",
-                                          "csc",
-                                          "floor",
-                                          "sec",
-                                          "pow",
-                                          "erfc",
-                                          "rand",
-                                          "rand_seed"};
-          for (const auto &function_name_c_string : function_names)
+          internal::MuParserFunctionNames function_names;
+          for (auto function_name_it = function_names.begin();
+        		  function_name_it != function_names.end();
+        		  ++function_name_it)
```

Comment:
Is there any reason not to use a range-based for loop here (as was done previously)?

Reply:
Changed it to range based loop.